include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -387,12 +387,16 @@ class operational_domain_impl
     {
         const auto logic_cells = lyt.get_cells_by_type(technology<Lyt>::cell_type::LOGIC);
 
-        if (!logic_cells.empty())
+        if (params.operational_params.strategy_to_analyze_operational_status ==
+            is_operational_params::operational_analysis_strategy::FILTER_ONLY)
         {
-            for (const auto& c : logic_cells)
-            {
-                canvas_lyt.assign_cell_type(c, technology<Lyt>::cell_type::NORMAL);
-            }
+            assert(logic_cells.size() > 0 && "No logic cells found in the layout");
+        }
```

Comment:
I would convert this `if` into one big `assert`. Because right now, this code reduces to an empty `if` statement potentially triggering a warning in `Release` mode.

Reply:
good point! thanks!